### PR TITLE
4 verifier

### PR DESCRIPTION
**Pull request by sgreenbury**
Completes #4.

### CODE REVIEW

File: trustchain-core/src/data.rs [hunk 1]
```diff
@@ -564,6 +564,19 @@ pub const TEST_ROOT_PLUS_1_DOCUMENT_METADATA: &str = r##"
...
+{
+   "kty": "EC",
+   "crv": "secp256k1",
+   "x": "7ReQHHysGxbyuKEQmspQOjL7oQUqDTldTHuc9V3-yso",
+   "y": "kWvmS7ZOvDUhF8syO08PBzEpEk3BZMuukkvEJOKSjqE"
+}
+"##;
+
+/// Proof value from metadata
+pub const TEST_ROOT_PLUS_1_JWT: &str = "eyJhbGciOiJFUzI1NksifQ.IkVpQXM5dkx2SmdaNkFHMk5XbUFmTnBrbl9EMlNSSUFSa2tCWE9kajZpMk84Umci.awNd-_O1N1ycZ6i_BxeLGV14ok51Ii2x9f1FBBCflyAWw773sqiHvQRGHIMBebKMnzbxVybFu2qUEPWUuRAC9g";
```
Comment (sgreenbury) [inline trustchain-core/src/data.rs:578]: Rename variable as JWS not JWT

File: trustchain-core/src/data.rs [hunk 2]
```diff
@@ -579,3 +592,5 @@ pub const TEST_ROOT_PLUS_2_DOCUMENT_METADATA: &str = r##"
    }
 }
 "##;
+/// Proof value from metadata
+pub const TEST_ROOT_PLUS_2_JWT: &str = "eyJhbGciOiJFUzI1NksifQ.IkVpQTNtT25QRklDbTdyc2ljVjRIaFMtNjhrT21xMndqa2tlMEtkRnkzQWlWZlEi.Fxlbm8osH2O5KOQ9sS21bypT_WoWxVD8toCU4baBnLk_gOxiOy_n3cMFMVANJ8usPrKAfRFeC27ATTkWBYZzuw";
```
Comment (sgreenbury) [inline trustchain-core/src/data.rs:596]: Rename as JWS not JWT

**Comment by sgreenbury** [comment]
- [ ] Remove the `test_chain()` function and instead make a `str` version importable from `data.rs` for use both within crate and external to the crate

**Comment by sgreenbury** [comment]
Update Trustchain proof service ID and type with lib consts.

File: trustchain-core/src/utils.rs [hunk 3]
```diff
@@ -33,11 +75,40 @@ pub fn get_did_suffix(did: &str) -> &str {
...
+/// Extracts payload from JWT and verifies signature.
+pub fn decode_verify(jwt: &str, key: &JWK) -> Result<String, ssi::error::Error> {
+    ssi::jwt::decode_verify(jwt, key)
+}
+
+/// Extracts and decodes the payload from the JWT.
+pub fn decode(jwt: &str) -> Result<String, ssi::error::Error> {
+    ssi::jwt::decode_unverified(jwt)
+}
+
```
Comment (sgreenbury) [inline trustchain-core/src/utils.rs:111]: Change to JWS

File: trustchain-core/src/utils.rs [hunk 4]
```diff
@@ -50,14 +121,36 @@ pub fn set_panic_hook() {
...
+        Ok(())
+    }
+
+    #[test]
+    fn test_decode_canonicalize_hash() -> Result<(), Box<dyn std::error::Error>> {
+        let doc: Document = serde_json::from_str(TEST_ROOT_PLUS_1_DOCUMENT)?;
+        let doc_canon = canonicalize(&doc)?;
+        let actual_hash = hash(&doc_canon);
+        let jwt = TEST_ROOT_PLUS_1_JWS;
+        let expected_hash = decode(jwt)?;
```
Comment (sgreenbury) [inline trustchain-core/src/utils.rs:150]: jws

File: trustchain-core/src/utils.rs (hunk 3 above)
Comment (sgreenbury) [reply trustchain-core/src/utils.rs:111]: Due to the proof mechanics currently implemented as JWT, postpone to #61

File: trustchain-core/src/utils.rs (hunk 4 above)
Comment (sgreenbury) [reply trustchain-core/src/utils.rs:150]: Due to the proof mechanics currently implemented as JWT, postpone to #61

**Review by thobson88 — APPROVED** [review]
LGTM

**Pull request merged**